### PR TITLE
txn: use thread pool for scheduler.

### PR DESCRIPTION
**Pull request by disksing**
@zhangjinpeng1987 @BusyJay

### CODE REVIEW

File: src/storage/mod.rs [hunk 1]
```diff
@@ -172,36 +170,17 @@ impl fmt::Debug for Command {
...
-                    Message::Command(cmd) => scheduler.handle_cmd(cmd),
-                    Message::Close => break,
-                }
-            }
-            info!("storage: [{}] closing.", desc);
-            Ok(())
-        }));
-
+        let sched = Scheduler::new(engine.clone());
+        info!("storage {} started.", format!("{:?}", engine));
```
Comment (siddontang) [inline src/storage/mod.rs]: why format?

**Comment by siddontang** [comment]
LGTM

File: src/storage/mod.rs [hunk 2]
```diff
@@ -211,20 +190,23 @@ impl Storage {
...
 
     pub fn stop(&mut self) -> Result<()> {
-        if self.thread.is_none() {
-            return Ok(());
-        }
-        try!(self.tx.send(Message::Close));
-        if self.thread.take().unwrap().join().is_err() {
-            return Err(box_err!("failed to wait storage thread quit"));
-        }
+        self.sched.take();
```
Comment (BusyJay) [inline src/storage/mod.rs]: Do we need to shutdown the `threadpool`?

Comment (disksing) [inline src/storage/mod.rs]: @BusyJay The threadPool does not supply a `shutdown` method. Instead all threads will quit after the pool is dropped. Refer https://frewsxcv.github.io/rust-threadpool/src/threadpool/lib.rs.html#367

**Comment by BusyJay** [comment]
LGTM

**Comment by disksing** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**